File: Assignment40/program2.c
//////////////////////////////////////////////////////////////////////
// File name :       program2.c
// Discription :     It is Recusive program to print certain pattern.
// Author :          Mangesh Ashok Bedre
// Date :            1/09/2025
//
//////////////////////////////////////////////////////////////////////

#include<stdio.h>

//////////////////////////////////////////////////////////////////////
// Function name :   Display
// Discription :     It is used to print pattern.
// Input  :          Nothing
// Output :          1   2   3   4   5
// Author :          Mangesh Ashok Bedre
// Date :            1/09/2025
//
//////////////////////////////////////////////////////////////////////

void Display()
{
    static int i = 1;
    
    if(i > 5)
    {
        return;
    }
    else
    {
        printf("%d\t",i);
        i++;
        Display();
    }
}

int main()
{
    Display();
    return 0;
}

File: Assignment48/program2.java
import java.util.Scanner;
//////////////////////////////////////////////////////////////////////
// File name :       program2.c
// Discription :     It is a program to print  pattern.
// Author :          Mangesh Ashok Bedre
// Date :            1/09/2025
//
//////////////////////////////////////////////////////////////////////
class Pattern
{
    
    //////////////////////////////////////////////////////////////////////
    // Function Name :   Display
    // Discription :     It is used to print pattern.
    // Input  :          String
    // Output :          H e l l o
    //                   H e l l
    //                   H e l
    //                   H e
    //                   H
    // Author :          Mangesh Ashok Bedre
    // Date :            1/09/2025
    //
    //////////////////////////////////////////////////////////////////////
    public void Display(String str)
    {
        int i = 0;
        int j = 0;

        int iCnt = 0;

        for(i = 0; i < str.length() ; i++)
        {
            for(j = str.length()-1 , iCnt = 0 ;j >= 0 - i; j--,iCnt++)
            {
                if(i <= j)
                {
                    System.out.print(str.charAt(iCnt)+"\t");
                }
                else{
                    System.out.print("\t");
                }
            }
            System.out.println();
        }
    }
}

class program2
{
    public static void main(String a[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the String : ");
        String str = sobj.nextLine();

        Pattern pobj = new Pattern();
        pobj.Display(str);
    }
}